public/app/features/transformers/regression/regression.ts

```diff
@@ -0,0 +1,112 @@
+import { PolynomialRegression } from 'ml-regression-polynomial';
+import { SimpleLinearRegression } from 'ml-regression-simple-linear';
+import { map } from 'rxjs';
+
+import {
+  DataFrame,
+  DataTransformerID,
+  FieldMatcherID,
+  FieldType,
+  SynchronousDataTransformerInfo,
+  fieldMatchers,
+} from '@grafana/data';
+
+export enum ModelType {
+  linear = 'linear',
+  polynomial = 'polynomial',
+}
+
+export interface RegressionTransformerOptions {
+  modelType?: ModelType;
+  degree?: number;
+  xFieldName?: string;
+  yFieldName?: string;
+  resolution?: number;
+}
+
+export const DEFAULTS = { resolution: 100, modelType: ModelType.linear, degree: 2 };
+
+export const RegressionTransformer: SynchronousDataTransformerInfo<RegressionTransformerOptions> = {
+  id: DataTransformerID.regression,
+  name: 'Regression',
+  operator: (options, ctx) => (source) =>
+    source.pipe(map((data) => RegressionTransformer.transformer(options, ctx)(data))),
+  transformer: (options, ctx) => {
```

Comment:
I really like the implementation of this transformation! Great job with it!  :tada::tada: